src/connectionconfig/azureHelper.ts

```diff
@@ -97,3 +107,76 @@ export async function getQuickPickItems(
 
     return quickPickItems;
 }
+
+const serverResourceType = "Microsoft.Sql/servers";
+const databaseResourceType = "Microsoft.Sql/servers/databases";
+const elasticPoolsResourceType = "Microsoft.Sql/servers/elasticpools";
+
+export async function fetchServersFromAzure(
+    sub: AzureSubscription,
+): Promise<AzureSqlServerInfo[]> {
+    const result: AzureSqlServerInfo[] = [];
+    const client = new ResourceManagementClient(
+        sub.credential,
+        sub.subscriptionId,
+    );
+
+    // for some subscriptions, supplying a `resourceType eq 'Microsoft.Sql/servers/databases'` filter to list() causes an error:
+    // > invalid filter in query string 'resourceType eq "Microsoft.Sql/servers/databases'"

```

Comment:
Is it possible to fall back to client-side filtering only if Azure API side filtering is failing with this error>

Reply:
Yeah, I thought about doing that, but in my light testing on a subscription with 460+ total resources where only 75 were databases, the performance difference is negligible.  There seems to be a high fixed cost of querying the Azure management plane, and the number of resources returned doesn't appear to impact it as much, so filtering doesn't have a big benefit.

I have a telemetry point for how long it takes to load the Azure info that we can look at to see if it's worth doing perf testing to optimize further here.